app/controllers/posts_controller.rb

```diff
@@ -1,23 +1,58 @@
 class PostsController < ApplicationController
-  before_action :find_user, only: %i[index show]
-  before_action :find_post, only: [:show]
+  before_action :find_user, only: %i[show new create]
+  before_action :find_post, only: %i[show]
 
+  # GET /posts or /users/:user_id/posts
   def index
-    @posts = @user.posts
+    if params[:user_id]
+      @user = User.find_by(id: params[:user_id])
+      if @user
+        @posts = @user.posts
+      else
+        flash[:alert] = 'User not found.'
+        redirect_to root_path and return
+      end
+    else
+      @posts = Post.all
+    end
   end
 
+  # GET /users/:user_id/posts/:id
   def show; end
 
+  # GET /users/:user_id/posts/new
+  def new
+    @post = @user.posts.build
+  end
+
+  # POST /users/:user_id/posts
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
+
   private
 
   def find_user
     @user = User.find(params[:user_id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = 'User not found.'
+    redirect_to root_path
```

Comment:
[OPTIONAL]
- [ ] According to the project requirement a post should be created on behalf of the current user. So kindly make use of the `current_user` method that you created in your `ApplicationController` to define the user.
```rb
@user = current_user
```
_**This is a requirement of the project but the first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change**_

Reply:
@LionRouge1 
I made the change. Thankyou for this suggestion 👍🏻